library/farmosUtil/farmosUtil.soilDisturbanceTerminationLog.unit.cy.js

```diff
@@ -0,0 +1,630 @@
+                    'Creating soil disturbance log should have failed.'
+                  );
+                })
+                .catch((error) => {
+                  expect(error.message).to.equal(
+                    'Request failed with status code 401'
+                  );
+                })
+            );
+          }
+        );
+      });
+
+      // Recheck the plant asset to ensure its state remains intact
+      cy.get('@newPlantAsset').then((plantAsset) => {
+        cy.wrap(farmosUtil.getPlantAsset(plantAsset.id)).then(
+          (updatedAsset) => {
+            expect(updatedAsset.relationships.location).to.have.length(3); // Beds should remain intact
+            expect(updatedAsset.attributes.status).to.not.equal('archived'); // Asset should not be archived
+          }
+        );
+      });
+
+      // Delete the movement log and plant asset
+      cleanupLogsAndAsset({
+        deleteMovementLog: true,
+        deletePlantAsset: true,
+      });
+    }
+  );
+
+  it(
+    'Error creating a soil disturbance termination log for a plant asset with no beds',
+    { retries: 4 },
+    () => {
+      createPlantAsset(
+        'Test Plant Asset with No Beds',
+        'HERB-CILANTRO',
+        '2023-11-20'
+      );
+
+      // Simulate an error while creating the soil disturbance termination log
+      cy.intercept('POST', '**/api/log/activity', {
+        statusCode: 401,
+      });
+
+      cy.get('@newPlantAsset').then((plantAsset) => {
+        cy.wrap(farmosUtil.getPlantAsset(plantAsset.id)).then(
+          (updatedPlantAsset) => {
+            cy.wrap(
+              farmosUtil
+                .createSoilDisturbanceTerminationLog(
+                  '2023-11-20',
+                  'CHUAU', // Location
+                  [], // No beds to terminate
+                  updatedPlantAsset // Pass the plant asset with no beds
+                )
+                .then(() => {
+                  throw new Error(
+                    'Creating soil disturbance log for plant with no beds should have failed.'
+                  );
+                })
+                .catch((error) => {
+                  expect(error.message).to.equal(
+                    'Request failed with status code 401'
+                  );
+                })
+            );
+          }
+        );
+      });
+
+      // Recheck the plant asset to ensure its state remains intact
+      cy.get('@newPlantAsset').then((plantAsset) => {
+        cy.wrap(farmosUtil.getPlantAsset(plantAsset.id)).then(
+          (updatedAsset) => {
+            expect(updatedAsset.relationships.location).to.have.length(0); // No beds should exist
+            expect(updatedAsset.attributes.status).to.not.equal('archived'); // Asset should not be archived
+          }
+        );
+      });
+
+      // Delete the plant asset
+      cleanupLogsAndAsset({
+        deletePlantAsset: true,
+      });
+    }
+  );
+
+  it('Delete a soil disturbance termination log', () => {
```

Comment:
We probably don't need this test since the `cleanupLogAndAsset` function check for the deletion.